src/Batcave/main.ts

```
@@ -0,0 +1,785 @@
+                });
+            }
+        } catch (err) {
+            console.error("Error parsing JSON data:", err);
+        }
+
+        return chapters;
+    }
+
+    async getChapterDetails(chapter: Chapter): Promise<ChapterDetails> {
+        try {
+            const request = {
+                url: `${baseUrl}/reader/${chapter.sourceManga.mangaId.split("-")[0]}/${chapter.chapterId}`,
+                method: "GET",
+            };
+
+            const $ = await this.fetchCheerio(request);
+
+            const pages: string[] = [];
+
+            const scriptData = $("script")
+                .filter((_, el) => $(el).html()?.includes("__DATA__") ?? false)
+                .first()
+                .html();
+
+            if (scriptData) {
+                const jsonMatch = scriptData.match(
+                    /window\.__DATA__\s*=\s*({[\s\S]*?})\s*;/,
+                );
+                if (jsonMatch) {
+                    try {
+                        const data: { images?: string[] } = JSON.parse(
+                            jsonMatch[1],
+                        ) as {
+                            images?: string[];
+                        };
+
+                        if (data.images && Array.isArray(data.images)) {
+                            data.images = data.images.map((img: string) =>
+                                img.replace(/\\\//g, "/"),
+                            );
+                            pages.push(...data.images);
+                        } else {
+                            console.error("Images not found in JSON data");
+                        }
+                    } catch (error) {
+                        console.error("Failed to parse JSON:", error);
+                    }
+                }
+            }
+
+            return {
+                id: chapter.chapterId,
+                mangaId: chapter.sourceManga.mangaId,
+                pages: pages,
+            };
+        } catch (error) {
```

You would want to throw an actual error instead of catching it and returning an empty list.